### PR TITLE
[FEATURE] Improvement - char limit on tags

### LINKED ISSUE #9345: [FEATURE] Improvement - char limit on tags
### Description

Currently there is no limit to how long a tag can be on a profile.  To ensure a consistent experience with tags, this issue is to provide a character limit on tags inputs/save for profile tags.

This is a multi-part issue that should include the following:
1. Add a character limit of 32 on a tag
2. In account management - the field description for tags should be updated to inform the user of the character limit
3. An alert should be presented to the user if they exceed the character limit on a single tag and should prevent saving of longer tags
4. Documentation should be updated to include the character limit on tags

### Screenshots

_No response_

### Do you want to work on this issue?

No

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by pratik9818**
## Fixes Issue 

closes #9345

## Check List (Check all the applicable boxes) 

- [ x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ x] I have updated the documentation accordingly.
- [ x] All new and existing tests passed.
- [x ] This PR does not contain plagiarized content.
- [ x] The title of my pull request is a short description of the requested changes.

## Screenshots

Screenshot 2023-11-09 233120

Screenshot 2023-11-10 234926
Screenshot 2023-11-10 235013

### CODE REVIEW

**Comment by akash19coder** [comment]
@pratik9818 I have noticed there are so many formatting issues in the files where you have made changes. Please format your code and push the formatted code🙂.

File: components/tag/TagsInput.js [hunk 1]
```diff
@@ -1,20 +1,32 @@
...
+  const maxTagChar = 32; 
   const handleKeyUp = (e) => {
     const inputValue = inputRef.current.value;
-    if (e.keyCode === comma || inputValue.endsWith(",")) {
-      const newTag = inputValue.trim().replace(/,/g, "");
+    if(inputValue.length >= maxTagChar && e.keyCode !== backspace){  // '=' sign because in 32 char ',' sign does't count
+      setShowNotification({
+        show: true,
+      type: "error",
+      message: "Tag",
```
Comment (akash19coder) [inline components/tag/TagsInput.js:22]: We can say "Tag LImit Exceeded" in the message?

Comment (accodes21) [reply components/tag/TagsInput.js:22]: Yeah it would be more expressive and helpful

File: pages/docs/how-to-guides/editing-forms.mdx [hunk 2]
```diff
@@ -28,11 +28,13 @@ You can change any of the form fields except `username` as this will always show
...
-Note: as you complete these fields the information will appear on the right hand side, which will allow you to preview how your Profile will look.
+5. Maximum 32 characters allowed in each tag , In case you exceed the characters than you can not separate or save the tag by comma .
 
-5. Click `Save`
+Note: as you complete these fields the information will appear on the right hand side, which will allow you to preview how your Profile will look.  
 
-6. Your changed Profile will be then be available on your custom URL (BioDrop.io/eddiejaoude)
+6. Click `Save`
+
+7. Your changed Profile will be then be available on your custom URL (BioDrop.io/eddiejaoude)
```
Comment (accodes21) [inline pages/docs/how-to-guides/editing-forms.mdx:37]: Minor grammatical correction: ```Your changed profile will then be available at your custom URL```

**Review by accodes21 — COMMENTED** [review]
Looks good

File: components/tag/TagsInput.js (hunk 1 above)
Comment (pratik9818) [reply components/tag/TagsInput.js:22]: ok

File: pages/docs/how-to-guides/editing-forms.mdx (hunk 2 above)
Comment (pratik9818) [reply pages/docs/how-to-guides/editing-forms.mdx:37]: ok

File: components/tag/TagsInput.js [hunk 3]
```diff
@@ -1,20 +1,32 @@
 import { useRef } from "react";
 import XMarkIcon from "@heroicons/react/20/solid/XMarkIcon";
 import Input from "../form/Input";
-
-export default function TagsInput({ tags, onTagAdd, onTagRemove }) {
+import Notification from "@components/Notification";
```
Comment (akash19coder) [inline components/tag/TagsInput.js:4]: How about using 'Alert' component here? Wouldn't it make more sense here since we are alerting user that tag limit has exceeded?

Comment (pratik9818) [reply components/tag/TagsInput.js:4]: Yes, I can use the alert component, but the problem is that when I use the alert component, it appears just above the input tag, not in the upper right corner like the Notification component. To achieve this, I have to make changes to the CSS of the alert component. I think that would not be a great idea to change the CSS or add some condition, as other components might be using the alert component.

**Comment by pratik9818** [comment]
hi @SaraJaoude , I completely understand you're busy.  I would be grateful if you could take some time to review the pull request  .I look forward to hearing your thoughts on the changes if something needs to add or remove in code .

File: cls [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+core.logallrefupdates=true
+core.symlinks=false
+core.ignorecase=true
+core.hookspath=.husky
+remote.origin.url=https://github.com/pratik9818/BioDrop.git
+remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*
+branch.main.remote=origin
+branch.main.merge=refs/heads/main
+branch.char.remote=origin
+branch.char.merge=refs/heads/char
```
Comment (eddiejaoude) [inline cls:13]: what is this file for?

File: pages/docs/how-to-guides/editing-forms.mdx [hunk 5]
```diff
@@ -28,11 +28,13 @@ You can change any of the form fields except `username` as this will always show
 
 ![BioDrop Forms page](https://github.com/EddieHubCommunity/BioDrop/assets/82668196/86119a20-412a-449a-acba-87df2510ad25)
 
-Note: as you complete these fields the information will appear on the right hand side, which will allow you to preview how your Profile will look.
+5. Maximum 32 characters allowed in each tag , In case you exceed the characters than you can not separate or save the tag by comma .
```
Comment (eddiejaoude) [inline pages/docs/how-to-guides/editing-forms.mdx:31]: Thank you 👍

File: pages/account/manage/profile.js [hunk 6]
```diff
@@ -243,9 +243,11 @@ export default function Profile({ BASE_URL, profile, fileExists }) {
                         onTagAdd={handleTagAdd}
                         onTagRemove={handleTagRemove}
                         tags={tags}
+                        showNotification = {showNotification}
+                        setShowNotification = {setShowNotification}
                       />
                       <p className="text-sm text-primary-medium-low dark:text-primary-low-high">
-                        Separate tags with commas.
+                        Separate tags with commas and Maximum 32 characters allowed in each tag.
```
Comment (eddiejaoude) [inline pages/account/manage/profile.js:250]:
```suggestion
                        Separate tags with commas and they have a maximum of 32 characters allowed in each tag.
```

File: components/tag/TagsInput.js [hunk 7]
```diff
@@ -1,8 +1,14 @@
...
```
Comment (eddiejaoude) [inline components/tag/TagsInput.js:4]:
```suggestion
import Notification from "@components/Notification";

```

File: components/tag/TagsInput.js [hunk 8]
```diff
@@ -11,10 +18,22 @@ export default function TagsInput({ tags, onTagAdd, onTagRemove }) {
     backspace: 8,
   };
 
+  const maxTagChar = 32;
```
Comment (eddiejaoude) [inline components/tag/TagsInput.js:21]: I think this can be renamed to `maxLength` because it is already in the tag component

File: components/tag/TagsInput.js [hunk 9]
```diff
@@ -11,10 +18,22 @@ export default function TagsInput({ tags, onTagAdd, onTagRemove }) {
...
+        type: "error",
+        message: "Tag",
+        additionalMessage: "Tag limit exceeded",
+      });
+      return;
+    }
+
     if (e.keyCode === comma || inputValue.endsWith(",")) {
-      const newTag = inputValue.trim().replace(/,/g, "");
+      let newTag = inputValue.trim().replace(/,/g, "");
```
Comment (eddiejaoude) [inline components/tag/TagsInput.js:36]: why is this changed required?

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Looks good 👍  I have left inline comments

Also it breaks on the manage event page add/edit with tags

Screenshot 2023-11-17 at 18 23 36

File: cls (hunk 4 above)
Comment (pratik9818) [reply cls:13]: ok

File: components/tag/TagsInput.js (hunk 8 above)
Comment (pratik9818) [reply components/tag/TagsInput.js:21]: ok

**Comment by pratik9818** [comment]
@eddiejaoude , should i update the tag description in event page as well , like a profile page
Screenshot 2023-11-18 003534

**Comment by eddiejaoude** [comment]
> eddiejaoude , should i update the tag description in event page as well , like a profile page 

yes please make it consistent

Can you also resolve the conflicts please

**Comment by eddiejaoude** [comment]
Did you mean to close the PR? 

No need to force push, commits will be squashed, plus this makes reviewing very difficult

**Comment by pratik9818** [comment]
i did not do nothing, i don't know how it happend , what should i do now ?

**Comment by eddiejaoude** [comment]
You can see from the timeline in the issue above your comment that you closed and forced pushed to your branch

Screenshot 2023-11-17 at 19 37 24

I am not sure, I never force push. Try to reopen the PR

**Comment by pratik9818** [comment]
can i open the another PR because i am not able to reopen it

**Comment by pratik9818** [comment]
reopen this PR

**Comment by pratik9818** [comment]
@eddiejaoude , i reopend the PR

**Comment by eddiejaoude** [comment]
Oh your force push undid my changes I made 😞  - I will look to make them again soon

**Comment by pratik9818** [comment]
Apology @eddiejaoude , i will keep in mind next time ..

**Comment by eddiejaoude** [comment]
> Apology @eddiejaoude , i will keep in mind next time . But i made the all changes you said to changed

no problem, I made additional changes in addition to the comments

**Comment by pratik9818** [comment]
hi @eddiejaoude what should i do now , i mean, is their any changes i have to do ?

File: components/tag/TagsInput.js [hunk 10]
```diff
@@ -1,19 +1,37 @@
+import { useRef } from "react";
 import XMarkIcon from "@heroicons/react/20/solid/XMarkIcon";
 import Input from "../form/Input";
-
-export default function TagsInput({ tags, onTagAdd, onTagRemove, inputRef }) {
-
+import Notification from "@components/Notification";
```
Comment (eddiejaoude) [inline components/tag/TagsInput.js:4]:
```suggestion
import Notification from "@components/Notification";

```

File: pages/account/manage/event/[[...data]].js [hunk 11]
```diff
@@ -340,10 +336,11 @@ export default function ManageEvent({ BASE_URL, event }) {
                       onTagAdd={handleTagAdd}
                       onTagRemove={handleTagRemove}
                       tags={tags}
-                      inputRef={tagInputRef}
+                      showNotification = {showNotification}
+                        setShowNotification = {setShowNotification}
                     />
                     <p className="text-sm text-primary-medium-low dark:text-primary-low-high">
-                      Separate tags with commas.
+                      Separate tags with commas and Maximum 32 characters allowed in each tag.
```
Comment (eddiejaoude) [inline pages/account/manage/event/[[...data]].js:343]:
```suggestion
                      Separate tags with commas (max 32 characters).
```

File: pages/account/manage/event/[[...data]].js [hunk 12]
```diff
@@ -340,10 +336,11 @@ export default function ManageEvent({ BASE_URL, event }) {
                       onTagAdd={handleTagAdd}
                       onTagRemove={handleTagRemove}
                       tags={tags}
-                      inputRef={tagInputRef}
+                      showNotification = {showNotification}
+                        setShowNotification = {setShowNotification}
```
Comment (eddiejaoude) [inline pages/account/manage/event/[[...data]].js:340]: I think this indentation is not correct?

File: pages/account/manage/profile.js [hunk 13]
```diff
@@ -250,10 +243,11 @@ export default function Profile({ BASE_URL, profile, fileExists }) {
                         onTagAdd={handleTagAdd}
                         onTagRemove={handleTagRemove}
                         tags={tags}
-                        inputRef={tagInputRef}
+                        showNotification = {showNotification}
+                        setShowNotification = {setShowNotification}
                       />
                       <p className="text-sm text-primary-medium-low dark:text-primary-low-high">
-                        Separate tags with commas.
+                        Separate tags with commas and Maximum 32 characters allowed in each tag.
```
Comment (eddiejaoude) [inline pages/account/manage/profile.js:250]:
```suggestion
                        Separate tags with commas (max 32 characters).
```

File: pages/account/manage/event/[[...data]].js (hunk 12 above)
Comment (pratik9818) [reply pages/account/manage/event/[[...data]].js:340]: should i fix this and update the branch ?

Comment (eddiejaoude) [reply pages/account/manage/event/[[...data]].js:340]: yes please 👍

Comment (eddiejaoude) [reply pages/account/manage/event/[[...data]].js:340]: Prettier plugin in VScode will fix this automatically

File: pages/account/manage/profile.js (hunk 13 above)
Comment (pratik9818) [reply pages/account/manage/profile.js]: should i change this 'Maximum 32 characters allowed in each tag.' to this '(max 32 characters)'

Comment (eddiejaoude) [reply pages/account/manage/profile.js]: I already made the change, it is marked as resolved

File: pages/account/manage/event/[[...data]].js (hunk 12 above)
Comment (pratik9818) [reply pages/account/manage/event/[[...data]].js]: done

File: components/tag/TagsInput.js [hunk 14]
```diff
@@ -1,19 +1,38 @@
...
+  onTagAdd,
+  onTagRemove,
+  showNotification,
+  setShowNotification,
+}) {
+  const inputRef = useRef(null);
 
   //key code
-  const { comma, backspace, enter } = {
+  const { comma, backspace } = {
```
Comment (eddiejaoude) [inline components/tag/TagsInput.js:16]: why was `enter` removed?

File: components/tag/TagsInput.js [hunk 15]
```diff
@@ -1,19 +1,38 @@
...
     comma: 188,
     backspace: 8,
-    enter: 13,
   };
 
+  const maxLength = 32;
   const handleKeyUp = (e) => {
     const inputValue = inputRef.current.value;
-    if (e.keyCode === comma || inputValue.endsWith(",") || e.keyCode === enter) {
+    if (inputValue.length >= maxLength && e.keyCode !== backspace) {
```
Comment (eddiejaoude) [inline components/tag/TagsInput.js:24]: why was `enter` removed?

File: pages/account/manage/profile.js [hunk 16]
```diff
@@ -247,8 +247,7 @@ export default function Profile({ BASE_URL, profile, fileExists }) {
                         setShowNotification={setShowNotification}
                       />
                       <p className="text-sm text-primary-medium-low dark:text-primary-low-high">
-                        Separate tags with commas and Maximum 32 characters
-                        allowed in each tag.
+                        Separate tags with commas (max 32 characters).
```
Comment (eddiejaoude) [inline pages/account/manage/profile.js:250]: why do my inline changes keep getting removed?

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Package lock file has been added, please remove

**Comment by pratik9818** [comment]
i am working on those issues

**Comment by eddiejaoude** [comment]
Ok if you need any help let us know

**Comment by pratik9818** [comment]
@eddiejaoude can i restart my char branch like updating char branch same as this project main branch because i think i am unable to update or back all those changes which u mentioned, can u help me out from this.

**Comment by pratik9818** [comment]
Status - still fixing it.

**Comment by pratik9818** [comment]
@eddiejaoude i fixed the issue and i had to hard push code for solveing that issue . kindly can you check ?
extremely sorry for your trouble .

File: components/tag/TagsInput.js [hunk 17]
```diff
@@ -1,19 +1,40 @@
 import XMarkIcon from "@heroicons/react/20/solid/XMarkIcon";
 import Input from "../form/Input";
+import Notification from "@components/Notification";
```
Comment (eddiejaoude) [inline components/tag/TagsInput.js:3]:
```suggestion
import Notification from "@components/Notification";

```

File: pages/account/manage/profile.js [hunk 18]
```diff
@@ -251,9 +251,11 @@ export default function Profile({ BASE_URL, profile, fileExists }) {
                         onTagRemove={handleTagRemove}
                         tags={tags}
                         inputRef={tagInputRef}
+                        showNotification={showNotification}
+                      setShowNotification={setShowNotification}
```
Comment (eddiejaoude) [inline pages/account/manage/profile.js:255]: The indentation is not consistent here

**Comment by eddiejaoude** [comment]
> i fixed the issue and i had to hard push code for solveing that issue . kindly can you check ? extremely sorry for your trouble .

Looks good, thank you for fixing, but force push is probably what is causing all the problems. I left a tiny inline comment

**Review by eddiejaoude — APPROVED** [review]
Thank you 👍

**Comment by pratik9818** [comment]
> Thank you 👍 

Thank you @eddiejaoude for supporting
